File: nivel-02/ex-05/bhaskara.c
// Código para encontrar as raízes de uma 
// equação de 2° grau

#include <stdio.h>
#include <assert.h>
#include <math.h>

int delta (int a, int b, int c){
	return b*b - (4*a*c);
}

int x1 (int a, int b, int c){
	return (-b + sqrt(delta(a, b, c)))/2*a;
}
	
int x2 (int a, int b, int c){
	return (-b - sqrt(delta(a, b, c)))/2*a;
}

int main(){
	int a, b, c;
	a = 1;
	b = 0;
	c = -16;

	assert(64 == delta(a, b, c));
	assert(4 == x1(a, b, c));
	assert(-4 == x2(a, b, c));
}

File: nivel-02/ex-05/bhaskara.py
#-*- coding:utf-8 -*-

import math

def calcula_delta(a, b, c): 
	delta = (b*b)-(4*a*c)
	return delta

def calcula_x1(a, b, c):
	delta = calcula_delta(a, b, c)
	x1 = (-b + math.sqrt(delta))/2*a
	return x1

def calcula_x2(a, b, c):
	delta = calcula_delta(a, b, c)
	x2 = (-b - math.sqrt(delta)/2*a)
	return x2
	
a = 1
b = 0
c = -16

assert (64 == calcula_delta(a, b, c))
assert (4 == calcula_x1(a, b, c))
assert (-4 == calcula_x2(a, b, c))
